### ISSUE TITLE
Suggestion: Mod categories

### ISSUE THREAD

**Issue by Blazingfireflies** [tModLoader/tModLoader#245]
I noticed that there are a lot of mods on tmodloader, and it s hard to find the type of mod one is looking for just by searching in a bar, and scrolling throught all the mods scrolls way too fast, so it would be nice if perhaps mods could be part of categories to help people find what they are looking for.

some categories i would do are: Content (thorium, calamity, tremor), Utility (Recipe browser, Boss checklist, Cheat Sheet), Small additions (Morepotions, NPC vending machines), RPG/ Levels (learn by doing, experience and classes). 

there would be a thing at the top next to the search bar that let you choose the categories you want to search in.
this would make searching for mods easier, and if the scroll bar scrolled at a reasonable pace when viewing a lot of mods then that would make it even better.

**Comment by hamstar0** [comment]
Tags might work better than categories, since categories can be hard to fit onto some mods.

**Comment by Jofairden** [comment]
@hamstar0 that is true, but there can definitely be a certain amount of set categories. I'd rather browse some categories before tags where modders add tags like 'poop' on their mod. I'd say probably more than 80% of the mods available can fit into at least one category we'd come up with, fairly certain.

**Comment by Chicken-Bones** [comment]
Nah, just have a set of approved tags, or perhaps even let the community collaborate on tags by only showing tags which apply to mods by more than 20 authors or something. 

I think of tags like categories, except you can belong to multiple of them

**Comment by Jofairden** [comment]
@Chicken-Bones this is a good one. What I've seen mostly is either a or b which you already noted, but not for tags. Usually modders can attach any tags they wish to their submission, usually separated by space or comma (see youtube, github, nexus?). Categories however, I've seen them pre-defined mostly. This is probably the safest option.

**Comment by Jofairden** [comment]
I want to come back to this. After 0.11 we were planning to migrate all existing backend to a tML server, including database. We can begin up cleaning architecture, and 'API-ify' things. Once done, I can make a browser page with filtering options within a day.

**Comment by Solxanich** [comment]
As an aside, [Mod Helpers](https://forums.terraria.org/index.php?threads/mod-helpers.63670/) adds Tags to mods already, and has a nice style going already. Not sure when it added tags relative to this old issue.
Should we loop back in Hamstar0 after all this time, and implement their work as part of 1.4 tML formally (assuming they are willing)? There probably is already a fairly well-grown DB of tags from 1.3 that would carry forward well in that case. I imagine the code isn't too far from being implemented in tML either. Of course we should credit accordingly

WRT Jofairden: I imagine there would be still be opportunity to do architecture cleanup with and without?

**Comment by Mirsario** [comment]
There is no need to look into this, because we're currently planning to replace most of mod browser with steam workshop (yes, it will work on GOG.)

**Comment by hamstar0** [comment]
SW's categories are inferior.

**Comment by Mirsario** [comment]
k.

**Comment by Solxanich** [comment]
Correct me if I'm wrong, this is just a thought/theory.
The steam workshop implementation would be mod browser lookups and retrieves mod info + file from steam workshop and then displays in mod browser. IE steam workshop does all the backend load, manages reporting mods, etc. 
The tags Hamstar implemented is an overlay in mod browser referencing a secondary DB based on modName, and thus can be implemented independent to the SW change as a mod browser UI bonus (ie tells you more than Steam alone, and provides additional capabilities that make searching by category for all users readily available).

tl;dr we could probably still add it, and do SW, and they'd still both be meaningful improvements

**Comment by Solxanich** [comment]
To further elaborate why I'm pushing; 
When it comes to content moderation (such as what might be a part of a release on Chinese platform WeGame) the tag based structure implemented in Mod Helpers would be significantly easier to extend to filter out "blacklisted" mods when compared against S. From the looks of the implementation, we could easily handover a "WeGame-Blocked" tag to WeGame to moderate and be liable for and have that tag filter games for that platform with an existing system. Thus my favorite type of solution to moderation: make it technically feasible, and hand the ongoing work on to those who want it done and leave them accountable without having to fuss about tri-party interactions.

**Comment by Mirsario** [comment]
There are currently no plans for us to release tml on wegame, or even support it right now. Especially now that you mention it being a pain in relation to censoring. There were multiple discussions about tags and sw in discord, and I don't remember anything solid ever being said in those (other than sw bad, i.e. what hamstar said above.) This is better off being discussed after steam workshop is implemented, and with jofairden present.

**Comment by Solxanich** [comment]
Agreed, well said. Given the context, this issue is best deferred until Steam Workshop implementation is mostly figured out. I think it best to close this issue and create a properly relevant issue that isn't vague to if it's going to be relevant or not within the upcoming design environment.
For now, I've created a separate issue that re-organizes what this request was for under the topic of steam workshop and a future improvement, so that this one can be closed as already encompassed within a relevant design issue (likewise for possibly the report function issue?).

**Comment by cozykoko** [comment]
Isnt this implemented with the steam workshop now?

**Comment by Solxanich** [comment]
Good eye. 
Unfortunately it is yes and no. 
I see two features here - 
Categories/tags itself, which is implemented Workshop side. 
And then bringing those back in to tml mod browser is the other task here. And that hasn't happened, and not sure how feasible will be yet

**Comment by Solxanich** [comment]
Outstanding:
All the code up the browser supports tags now.
1.4.4 tag is used for filtering the mod browser.

If we want additional tag filtering, have to implement UI to select that in-game.
